lua/octo/init.lua

```diff
@@ -761,4 +762,66 @@ function M.apply_buffer_mappings(bufnr, kind)
   end
 end
 
+function M.show_summary()
+  if vim.bo.ft ~= "octo_issue" then
+    return
+  end
+
+  local _, repo2 = pcall(api.nvim_buf_get_var, 0, "repo")
+  if not repo2 then
+    return
+  end
+
+  local cursor = api.nvim_win_get_cursor(0)
+  local current_line = vim.fn.getline(".")
+  local different_repo_issue_pattern = "%s([^ /]+/[^ #]+)#(%d+)%s"
+  local same_repo_issue_pattern = "%s#(%d+)%s"
+  local start_col, end_col, repo, number = current_line:find(different_repo_issue_pattern)
+
+  if start_col and end_col and start_col <= cursor[2] and cursor[2] <= end_col then
```

Comment:
Reds complicated. What does this if-statement express? Sometimes it is worth to create local variable with a descriptive name and assign the boolean expression to it. That supports the reader to understand whats going on here.

Despite I don't know what you are doing here: must it be so complicated? Could you provide background information? In my head I have a simple structure like: get word under cursor, check if matches the pattern of an issue reference, fetch the issue and open the preview. All other details should be handled in smaller functions. This concept might be wrong because I don't know the details or you just have an alternative/better approach. But unfortunately I have no chance to understand it. :slightly_frowning_face: